src/base/bittorrent/torrentimpl.cpp

```diff
@@ -2370,6 +2370,11 @@ void TorrentImpl::setMetadata(const TorrentInfo &torrentInfo)
     });
 }
 
+void TorrentImpl::setSslCertificate(const QByteArray &certificate, const QByteArray &privateKey, const QByteArray &dhParams)
+{
+    m_nativeHandle.set_ssl_certificate_buffer(certificate.toStdString(), privateKey.toStdString(), dhParams.toStdString());
```

Comment:
Why not save certificate files on the disk to be reused after restart?

Reply:
This is actually a good idea

Reply:
Added the logic to save certificates on disk and to clean up when the torrent is removed from the session.